src/PIL/ImageFile.py

```diff
@@ -63,15 +64,25 @@
 # Helpers
 
 
-def raise_oserror(error):
+def _get_oserror(error, *, encoder=False):
     try:
         msg = Image.core.getcodecstatus(error)
     except AttributeError:
         msg = ERRORS.get(error)
     if not msg:
-        msg = f"decoder error {error}"
-    msg += " when reading image file"
-    raise OSError(msg)
+        msg = f"{'encoder' if encoder else 'decoder'} error {error}"
+    msg += f" when {'writing' if encoder else 'reading'} image file"
+    return OSError(msg)
+
+
+def raise_oserror(error):
+    deprecate(
+        "raise_oserror",
+        12,
+        action="It is only useful for translating error codes from internal "
+        "Pillow APIs.",
+    )
+    raise _get_oserror(error, encoder=False)
```

Comment:
I wouldn't feel the need to specify the default value - but I realise there may be different views on this.

Reply:
Yeah, I didn't at first.  I figure it provides some protection against copypaste errors down the road.  The consequence of an incorrect `encoder` value is easy to miss.